triggers/service/src/main/scala/com/digitalasset/daml/lf/engine/trigger/ServiceMain.scala

```diff
@@ -112,6 +115,24 @@ class Server(dar: Option[Dar[(PackageId, Package)]], jdbcConfig: Option[JdbcConf
   private def listRunningTriggers(jwt: Jwt): List[String] = {
     triggersByToken.getOrElse(jwt, Set()).map(_.toString).toList
   }
+
+  private def timeStamp(): String = {
+    DateTimeFormatter.ofPattern("yyyy-MM-dd_HH:mmss").format(LocalDateTime.now)
+  }
+
+  private def triggerStatus(t: RunningTrigger, msg: String): Unit = {
+    val id = t.triggerInstance
+    val entry = (timeStamp(), msg)
+    triggerLog += triggerLog.get(id).map(logs => id -> (entry :: logs)).getOrElse(id -> List(entry))
```

Comment:
This line is quite confusing. Could be simplified with `fold` or a more direct `match` I think, something like
```suggestion
    triggerLog += id -> triggerLog.get(id).fold(List(entry), logs => entry :: logs)
```

Reply:
Re-examined it in light of the switch to vector. In all truth, you might have to squint for a second but it's not that bad IMO.

Reply:
Just to be sure, `+=` overwrites an existing mapping right?

Reply:
AFAICT, the semantics are to add key, value pairs that are new and replace key, value pairs that are missing.